File: C/mario_pyramid/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // user input has to be between 1 & 8, both included
    int height;

    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    // check if input procedure is correct
    // printf("stored: %i\n",height);

    // build pyramid
    int i, j, dots;
    // Zeilen
    for (i = 1; i <= height; i++)
    {
        dots = height - i;
        while (dots > 0)
        {
            printf(" ");
            dots--;
        }

        // Spalten
        for (j = 1; j <= i; j++)
        {
            printf("#");
        }
        printf("\n");
    }
}

File: Python/mario_pyramid/mario.py
from cs50 import get_int

while True:
    height = get_int("Height: ")
    if height > 0 and height < 9:
        break

space = 1
# Zeilen
for i in range(height):
    for space in range(height-i-1): # because i is 0 at the beginning
        print(" ", end = "")

    for j in range(i+1):
        print("#", end = "")

    print()